### PR TITLE
Add `viz` install option

### PR DESCRIPTION
**Pull request by vtnate**
- Add an option to install visualization dependencies (matplotlib) if desired.
- Install the new `viz` option during CI so existing functionality continues
- Update the error/help message if that hasn't been done and a viz method is called
- Fix some typos my IDE found

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -39,3 +38,5 @@ testing =
     pytest >= 7.1.1
     pytest-cov >= 3.0.0
     tox >= 3.24.5
+viz =
```
Comment (mitchute) [inline setup.cfg:41]: @Myoldmopar should we prefer `extras` instead of `viz`?

**Comment by mitchute** [comment]
@vtnate there's another reference to matplotlib in the README that might need to be checked

File: setup.cfg (hunk 1 above)
Comment (Myoldmopar) [reply setup.cfg:41]: I think viz is good.  It allows the freedom to later add another separate optional install configuration if that's ever needed.

**Review by mitchute — APPROVED** [review]

**Comment by mitchute** [comment]
LGTM

**Pull request merged**